### PR TITLE
Add readonly in color, date, selectWithList, time input

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

**Review by tnagorra — COMMENTED** [review]
readOnly should hide all action buttons from input

File: components/Input/SelectInputWithList/index.js [hunk 1]
```diff
@@ -138,23 +140,20 @@ class SelectInputWithList extends React.PureComponent {
...
-                newValue.splice(index, 1);
-            }
+        if (index !== -1) {
+            newValue.splice(index, 1);
+        }
 
-            const objectValues = this.getObjectFromValue(options, newValue);
-            this.setState({ objectValues });
+        const objectValues = this.getObjectFromValue(options, newValue);
+        this.setState({ objectValues });
```
Comment (tnagorra) [inline components/Input/SelectInputWithList/index.js]: this setState will also be called at componentWillRecieve props so this should not be done before onChange

File: components/Input/SelectInputWithList/index.js [hunk 2]
```diff
@@ -209,7 +209,8 @@ class SelectInputWithList extends React.PureComponent {
             selectClassNames.push(styles.hasTopRightChild);
         }
 
-        const Item = hideRemoveFromListButton ? ListItem : DismissableListItem;
+        const isListItem = hideRemoveFromListButton || readOnly;
```
Comment (tnagorra) [inline components/Input/SelectInputWithList/index.js]: isListItem name doesn't make sense here. a separate variable is not necessary

File: components/Input/SelectInputWithList/index.js [hunk 3]
```diff
@@ -209,14 +210,18 @@ class SelectInputWithList extends React.PureComponent {
             selectClassNames.push(styles.hasTopRightChild);
         }
 
-        const Item = hideRemoveFromListButton ? ListItem : DismissableListItem;
+        const Item = (
```
Comment (tnagorra) [inline components/Input/SelectInputWithList/index.js:213]: It pains my eyes

File: components/Input/TabularSelectInput/index.js [hunk 4]
```diff
@@ -225,7 +229,10 @@ class TabularSelectInput extends React.PureComponent {
     }
 
     createTableHeaders = (tableHeaders) => {
-        const { hideRemoveFromListButton } = this.props;
+        const {
```
Comment (tnagorra) [inline components/Input/TabularSelectInput/index.js:232]: change farmer! change farmer!

File: components/Input/SelectInputWithList/index.js (hunk 1 above)
Comment (AdityaKhatri) [reply components/Input/SelectInputWithList/index.js]: done

File: components/Input/SelectInputWithList/index.js (hunk 2 above)
Comment (AdityaKhatri) [reply components/Input/SelectInputWithList/index.js]: yuss

File: components/Input/SelectInputWithList/index.js (hunk 3 above)
Comment (AdityaKhatri) [reply components/Input/SelectInputWithList/index.js:213]: baal ho

File: components/Input/TabularSelectInput/index.js (hunk 4 above)
Comment (AdityaKhatri) [reply components/Input/TabularSelectInput/index.js:232]: baal ho baal ho

**Pull request merged**